UnityServer/functions/src/model/UnityManager.ts

```diff
@@ -0,0 +1,183 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as firebase from "firebase-admin";
+import { UnityStatus } from "./UnityStatus";
+import * as serviceAccountPlay from "../keys/placeholderKey.json";
+var Verifier = require('google-play-billing-validator');
+
+const FIRESTORE_USERS_COLLECTION = 'users';
+const FIRESTORE_PURCHASES_COLLECTION = 'purchases';
+
+enum SkuType {
+  ONE_TIME = 'inapp',
+  SUBS = 'subs'
+}
+/* UnityManager is part of Model layer.
+ * It manages request of users from the unity sample game.
+ */
+export class UnityManager {
+  private usersCollectionReference: FirebaseFirestore.CollectionReference;
+  private purchasesCollectionReference: FirebaseFirestore.CollectionReference;
+  private verifier: any;
+
+
+  constructor(firebaseApp: firebase.app.App) {
+    this.usersCollectionReference = firebaseApp.firestore().collection(FIRESTORE_USERS_COLLECTION);
+    this.purchasesCollectionReference = firebaseApp.firestore().collection(FIRESTORE_PURCHASES_COLLECTION);
+    this.verifier = new Verifier({
+      email: serviceAccountPlay.client_email,
+      key: serviceAccountPlay.private_key
+    });
+  }
+
+  async registerUser(userId: string): Promise<UnityStatus> {
+    const valuesToSave = {
+      registered: true
+    };
+
+    return this.usersCollectionReference.doc(userId).set(valuesToSave, { merge: true })
+      .then(() => {
+        return new UnityStatus(true);
+      }).catch(error => {
+        return new UnityStatus(false, error);
+      });
+  }
+
+  async saveGameData(userId: string, gameData: string): Promise<UnityStatus> {
+    const valuesToSave = {
+      gameData: gameData
+    };
+
+    return this.usersCollectionReference.doc(userId).set(valuesToSave, { merge: true })
+      .then(() => {
+        return new UnityStatus(true);
+      }).catch(error => {
+        return new UnityStatus(false, error);
+      });
+  }
+
+  async getGameData(userId: string): Promise<UnityStatus> {
+    return this.usersCollectionReference.doc(userId).get()
+      .then(doc => {
+        if (doc.exists && doc.data() ?.gameData !== undefined && doc.data() ?.gameData !== "") {
+          return new UnityStatus(true, "", doc.data() ?.gameData);
+        }
+        return new UnityStatus(false, "User is not registered in the database");
+      }).catch(error => {
+        return new UnityStatus(false, error);
+      });
+  }
+
+  async checkSubscriptionPriceChange(userId: string): Promise<UnityStatus> {
```

Comment:
would be great if we could have some comments for this method to explain what's going on

Reply:
Sure